yt_dlp/extractor/rumble.py

```diff
@@ -236,6 +222,96 @@ def _real_extract(self, url):
         }
 
 
+class RumbleIE(InfoExtractor):
+    _VALID_URL = r'''(?x)
+        https?://(?:www\.)?rumble\.com/
+        (?P<id>v            # ids start with v
+            (?!ideos)       # exclude https://rumble.com/videos
+            [0-9a-zA-Z]{4,} # video page identifier (case insensitive)
+            (?:-[^&?#$/]*)? # usually followed by -<title>.html otherwise redirected
+        )
+        [^/]*$              # queries are allowed, but no slashes
+    '''
```

Comment:
Isn't this better?

```suggestion
    _VALID_URL = r'https?://(?:www\.)?rumble\.com/(?!videos)(?P<id>v\w+)[^/]*$'
```

Reply:
The extended notation helps me during development. But it can be compacted afterwards, if the regex is simple enough. Yet I prefer to have the title part in the id to see which page is currently processed. Rumble keeps them short enough.